hawk/app/assets/javascripts/module/primitives.js

```diff
@@ -2,300 +2,397 @@
+                      return [root[lstname][elemname]];
+                    } else {
+                      return root[lstname][elemname];
+                    }
+                  }
+                }
+              }
+              return [];
+            };
+
+            // display attrlists
+            // attrlist: {'name': 'value'}
+            // mapping: {'name': {'type', 'default', 'longdesc', 'values'}}
+            var pal = {};
+            var pam = {};
+            $.each(lookup(data.resource_agent, "parameters", "parameter"), function(i, v) {
+              var name = v.name;
+              var longdesc = "";
+              if ($.trim(v.longdesc)) {
+                longdesc = v.longdesc;
+              } else if ($.trim(v.shortdesc)) {
+                longdesc = v.shortdesc;
+              }
+              var type = "string";
+              var defvalue = "";
+              if ("content" in v && v.content) {
+                if ("type" in v.content) {
+                  type = v.content.type;
+                }
+                if ("default" in v.content) {
+                  defvalue = v.content["default"];
+                }
+              }
+              if (v.required == "1") {
+                pal[v.name] = defvalue;
+              }
+              pam[v.name] = {
+                type: type,
+                default: defvalue,
+                longdesc: v.shortdesc + ": " + v.longdesc
+              };
+              $("#helpentries").append($("#tmpl-helpentry").render({
+                name: name,
+                longdesc: longdesc,
+                default: defvalue
+              }));
+            });
+
+            $('form #paramslist').html($("#jstmpl-paramslist").render());
+            var paramslist = $('form #paramslist fieldset');
+            if (new_resource) {
+              paramslist.data('attrlist', pal);
+            }
+            paramslist.data('attrlist-mapping', pam);
+            paramslist.attrList();
+
+            // meta attributes are never mandated by the agent anyway
+            $('form #metalist').html($("#jstmpl-metalist").render());
+            var metalist = $('form #metalist fieldset');
+            if (new_resource) {
+              metalist.data('attrlist', {'target-role': 'Stopped'});
+            }
+            metalist.attrList();
+            $("#helpentries").append($("#tmpl-metahelp").render());
+
+            $('form #oplist').html($("#jstmpl-oplist").render());
+            var oplist = $('form #oplist fieldset');
+            oplist.data('oplist-actions', lookup(data.resource_agent, "actions", "action"));
+            oplist.opList({create: new_resource});
+
+            // hide help texts
+            $('[data-help-target]').each(function() {
+              $($(this).data('help-target')).hide();
+            });
+
+
+            // enable toggleables
+            $('form').toggleify();
+
+          },
+          error: function(xhr, status, msg) {
```

Comment:
'msg' is defined but never used.<br>'status' is defined but never used.<br>'xhr' is defined but never used.